src/test/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXTest.java

```diff
@@ -16,37 +23,102 @@
 import org.powermock.core.classloader.annotations.PowerMockIgnore;
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.rule.PowerMockRule;
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 import static com.segment.analytics.Analytics.LogLevel.VERBOSE;
-import static com.segment.analytics.Utils.createTraits;
+import static com.segment.analytics.android.integrations.optimizelyx.OptimizelyXIntegration.options;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.MockitoAnnotations.initMocks;
-import static org.powermock.api.mockito.PowerMockito.verifyStatic;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, sdk = 18, manifest = Config.NONE)
-@PowerMockIgnore({ "org.mockito.*", "org.robolectric.*", "android.*" })
-//@PrepareForTest({ Optimizely.class }) //
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class)
+@PowerMockIgnore({ "org.mockito.*", "org.roboelectric.*", "android.*" })
+@PrepareForTest({ OptimizelyClient.class })
 public class OptimizelyXTest {
 
   @Rule public PowerMockRule rule = new PowerMockRule();
-  @Mock Application application;
   @Mock Analytics analytics;
+  private OptimizelyXIntegration integration;
+  private OptimizelyClient client;
+  private OptimizelyManager manager;
 
   @Before public void setUp() {
-  }
+    initMocks(this);
+    PowerMockito.mock(OptimizelyClient.class);
 
-  @Test public void identify() {
+    client = mock(OptimizelyClient.class);
+    integration = new OptimizelyXIntegration(analytics, manager, new ValueMap().putValue("trackKnownUsers", false), Logger.with(VERBOSE));
   }
 
   @Test public void track() {
+    Properties properties = new Properties();
+    Traits traits = new Traits()
+            .putValue("userId", "123")
+            .putValue("anonymousId", "456");
+    integration.track(new TrackPayloadBuilder().properties(properties).traits(traits).event("event").build());
+
+    verify(client).track("event", "456", properties.toStringMap());
+  }
+
+  @Test public void trackKnownUsers() {
+    integration.trackKnownUsers = true;
+
+    Properties properties = new Properties();
+    Traits traits = new Traits()
+            .putValue("userId", "123");
+    integration.track(new TrackPayloadBuilder().properties(properties).traits(traits).event("event").build());
+
```

Comment:
You may have to do something like `when(client.isValid()).thenReturn(true);`

Reply:
Awaiting feedback from Prateek on the new approach in the main lib before completing tests - these tests are still valid for the previous version of the integration.